signals.go

```diff
@@ -43,6 +45,15 @@ func watchForSignals() {
 					log.Print(err)
 				}
 			}
+			if socket != "" && !strings.HasPrefix(socket, "@") {
+				// we've been listening on a named Unix socket, delete it
+				// before we exit so subsequent runs can re-bind the same
+				// socket path
+				err := os.Remove(socket)
```

Comment:
This will almost certainly error out in combination with `-setuid`, since the socket is created before dropping privileges.  But you don't need to be root to bind a Unix socket (as long as you have write access to the parent directory) so this situation should rarely arise.

Reply:
Maybe we should do it after dropping the privileges then?

Reply:
I did wonder about either delaying opening the socket until after setuid or just forbidding the use of both `-socket` and `-setuid` at the same time. Like I say the only valid use case for combining `-socket` with `-setuid` would be if you wanted to bind to a socket file in a directory to which the target uid doesn't have write access, in which case moving the bind to happen after privileges have been dropped will fail the same way as this cleanup would in the current design.

My gut feeling is that the cleanest option is to make it an error if both `-socket` and `-setuid` are set.  If you _need_ to run webhook as a regular unprivileged user but listen on a socket as root, then I have another PR ready to go that stacks on top of this one that would make `webhook` compatible with systemd socket activation - that would mean that systemd itself manages the socket (as root) and it can spawn `webhook` as an unprivileged user with the socket already open.

Reply:
Yeah, that makes sense.

Reply:
Ah, ok, I was assuming if you agreed with my analysis then I’d implement that change before you merged… I’ll prepare the systemd PR and roll this fix into that.